docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -2,4 +2,173 @@
 Developer Setup
 ***************
 
-TBD
+Creating a development shell
+============================
+
+To create an interactive development ``bash`` shell:
+
+
+1. Download, install, and activate the latest `Miniforge3`_ for your system. If an existing conda (Miniforge, Miniconda, Anaconda, etc.) installation is available and writable, you may activate that and skip the first 5 recipe steps below.
+2. Install the `condev`_ package into the base environment.
+3. In a clone of the `workflow-tools repository`_, create the development shell.
+
+
+This recipe uses the ``aarch64`` (64-bit ARM) Miniforge for Linux, and installs into ``$HOME/conda``. Adjust as necessary for your target system.
+
+.. code::
+
+   wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-aarch64.sh
+   bash Miniforge3-Linux-aarch64.sh -bfp ~/conda
+   rm Miniforge3-Linux-aarch64.sh
+   source ~/conda/etc/profile.d/conda.sh
+   conda activate
+   conda install -y -c maddenp condev
+   cd /to/your/workflow-tools/clone
+   make devshell
+
+
+.. _paragraph: 
+
+If the above is successful, you will be in a ``workflow-tools``
+development shell. See below for usage information. You may exit the
+shell with ``exit`` or ``ctrl-d``.
+
+Future ``make devshell`` invocations will be almost instantaneous, as
+the underlying virtual environment will already exist. In general, all
+source code changes will be immediately live in the development shell,
+subject to execution, test, etc. But some changes – especially to the
+contents of the ``recipe/`` directory, or to the ``src/setup.py`` module
+– may require recreation of the development shell. If you know this is
+needed, or when in doubt: Exit the development shell, run
+``conda env remove -n DEV-uwtools`` to remove the old environment, then
+run ``make devshell`` to recreate it.
+
+If your development shell misses any functionality you’re used to in
+your main shell, you can create a ``~/.condevrc`` file, which will be
+sourced by ``make devshell``. When in doubt, you might
+
+
+sourced by ``make devshell``. When in doubt, you might
+
+.. code::
```

Comment:
Something else that the converter had that I decided to remove, but am happy to add back in if we would like it is the option to specify what type of code we are adding here: 
Is it `sh`?
Is it `python`?

[Define a custom role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html) associated with a specific language.

Originally this was documented as `.. code:: sh`.

Reply:
This is `sh` code. Probably specifying the type will add some syntax highlighting, which might look either nice or distracting -- maybe see what you think.